### PR TITLE
ACMS-3417: Update Acquia CMS Telemetry as per Acquia Connector changes.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #[ACMS-3417](https://acquia.atlassian.net/browse/ACMS-3417)

**Proposed changes**
Update Acquia CMS telemetry as per Acquia Connector changes.

**Alternatives considered**
NA

**Testing steps**
Follow from ticket.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_common/src/AcquiaCmsCommonServiceProvider.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function register(ContainerBuilder $container) {
    * {@inheritdoc}
    */
   public function alter(ContainerBuilder $container) {
-    if ($container->hasDefinition('acquia_connector.telemetry')) {
+    if ($container->hasDefinition('acquia_connector.telemetry') && PHP_SAPI !== 'cli') {
```
Comment (apathak18) [inline modules/acquia_cms_common/src/AcquiaCmsCommonServiceProvider.php:42]: && PHP_SAPI !== 'cli' should be in onTerminateResponse method as here we are altering the service

Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/AcquiaCmsCommonServiceProvider.php]: done.

**Review by chandan-singh7929 — APPROVED** [review]

File: modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ class AcquiaConnectorTelemetryOverride extends AcquiaTelemetry {
   public function onTerminateResponse(KernelEvent $event) {
     // Check if telemetry opted,
     // then only trigger AcquiaTelemetry::onTerminateResponse($event).
-    if ($this->state->get('acquia_connector.telemetry.opted', TRUE)) {
+    if (PHP_SAPI !== 'cli' && \Drupal::service('state')->get('acquia_connector.telemetry.opted', TRUE)) {
```
Comment (vishalkhode1) [inline modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php:28]: @rajeshreeputra Can we use DI here. 
In `AcquiaCmsCommonServiceOverrider` class, probably we can pass arguments something like:
```
->addArgument(new Reference('state'))
```
and use it here.

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

**Review by vishalkhode1 — APPROVED** [review]

**Comment by vishalkhode1** [comment]
Merging PR as failing integrated existing site tests will be fixed as part of ACMS-3649.

**Pull request merged**